oci/layout/oci_transport.go

```diff
@@ -111,12 +90,19 @@ func NewReference(dir, image string) (types.ImageReference, error) {
 	if err != nil {
 		return nil, err
 	}
-	// This is necessary to prevent directory paths returned by PolicyConfigurationNamespaces
-	// from being ambiguous with values of PolicyConfigurationIdentity.
-	if strings.Contains(resolved, ":") {
-		return nil, errors.Errorf("Invalid OCI reference %s:%s: path %s contains a colon", dir, image, resolved)
+
+	if runtime.GOOS == "windows" {
+		// On Windows we must allow for a ':' as part of the path
+		if strings.Count(resolved, ":") > 1 {
+			return nil, errors.Errorf("Invalid OCI reference %s:%s: path %s contains more than one colon", dir, image, resolved)
+		}
+	} else {
+		if strings.Contains(resolved, ":") {
+			return nil, errors.Errorf("Invalid OCI reference %s:%s: path %s contains a colon", dir, image, resolved)
+		}
```

Comment:
One more thing, prompted by https://github.com/projectatomic/skopeo/issues/455 :

Nowadays we don’t treat `:` in policy scopes specially — but we _do_ treat `:` specially in the non-canonicalized input path, as output in `StringWithinTransport` and re-parsed in `ParseReference`. So, actually, this check for colons should, if I am not mistaken again, validate `dir`, not `resolved`.

(We originally didn’t validate `dir` because `image` was not allowed to contain colons, so `dir_with_colons:image_without_colons` was not ambiguous. After `image` was allowed colons, this became ambiguous and we should have started refusing such `dir` values at that time.

Conceptually independently, but at at about the same time, `image` was dropped from `PolicyConfiguration*`, and the `resolved` check was no longer necessary.)

Reply:
ahh, so check is against the unresolved path. ok